src/transformState.js

```diff
@@ -5,7 +5,46 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (let i = 0; i < actions.length; i++) {
+    const action = actions[i];
+
+    switch (action.type) {
+      case 'addProperties': {
+        const extraData = action.extraData;
```

Comment:
better to just destructuring `action` to avoid creating extra variables

Reply:
it's the same here, I didn't understand what you mean, can you give an example?
I don't understand the logic.

Reply:
Please, don't leave unresolved comments. If you still have any questions you could ask them in fe_chat in Slack